src/SldStyleParser.ts

```diff
@@ -292,188 +284,203 @@ export class SldStyleParser implements StyleParser<string> {
+    const namedLayerName = get(sldObject, 'StyledLayerDescriptor.NamedLayer.Name.#text');
     return userStyleTitle ? userStyleTitle
       : namedLayerName ? namedLayerName : '';
   }
 
   /**
-   * Creates a GeoStyler-Style StrMatchesFunctionFilterr from a SLD strMatches Function.
+   * Get the geostyler-style Filter from a SLD Rule.
+   *
+   * Currently only supports one Filter per Rule.
    *
-   * @param {object} sldFilter The SLD Filter
-   * @return {Filter} The GeoStyler-Style FunctionFilter
+   * @param sldRule The SLD Rule
+   * @return The geostyler-style Filter
    */
-  getStrMatchesFunctionFilterFromSldFilter(sldFilter: any): StrMatchesFunctionFilter {
-    const propertyName = _get(sldFilter, 'Function[0].PropertyName[0]._');
-    const literal = _get(sldFilter, 'Function[0].Literal[0]._');
-    const regex = new RegExp(literal);
-    return [
-      'FN_strMatches',
-      propertyName,
-      regex
-    ];
+  getFilterFromRule(sldRule: any[]): GsFilter | undefined {
+    const sldFilter = get(sldRule, 'Filter');
+    if (!sldFilter || sldFilter.length === 0) {
+      return;
+    }
+    const operator = Object.keys(sldFilter[0])?.[0];
+    if (!operator) {
+      return;
+    }
+    const filter = this.getFilterFromOperatorAndComparison(operator, sldFilter);
+    return filter;
   }
 
   /**
-   * Creates a GeoStyler-Style FunctionFilter from a SLD Function.
+   * Get the geostyler-style ScaleDenominator from a SLD Rule.
    *
-   * @param {object} sldFilter The SLD Filter
-   * @return {Filter} The GeoStyler-Style FunctionFilter
+   * @param sldRule The SLD Rule
+   * @return The geostyler-style ScaleDenominator
    */
-  getFunctionFilterFromSldFilter(sldFilter: any): FunctionFilter {
-    const functionName = _get(sldFilter, 'Function[0].$.name');
-    switch (functionName) {
-      case 'strMatches':
-        return this.getStrMatchesFunctionFilterFromSldFilter(sldFilter);
-      default:
-        return this.getStrMatchesFunctionFilterFromSldFilter(sldFilter);
+  getScaleDenominatorFromRule(sldRule: any[]): ScaleDenominator | undefined {
+    const scaleDenominator: ScaleDenominator = <ScaleDenominator> {};
+    const min = get(sldRule, 'MinScaleDenominator.#text');
+    if (min) {
+      scaleDenominator.min = Number(min);
+    }
+    const max = get(sldRule, 'MaxScaleDenominator.#text');
+    if (max) {
+      scaleDenominator.max = Number(max);
     }
+
+    return (scaleDenominator.min || scaleDenominator.max)
+      ? scaleDenominator
+      : undefined;
+  }
+
+  /**
+   * Get the geostyler-style Symbolizers from a SLD Rule.
+   *
+   * @param sldRule The SLD Rule
+   * @return The geostyler-style Symbolizer array
+   */
+  getSymbolizersFromRule(sldRule: any[]): Symbolizer[] {
+    const symbolizers: Symbolizer[] = sldRule
+      .filter(isSymbolizer)
+      .map((sldSymbolizer: any) => {
+        const sldSymbolizerName: string = Object.keys(sldSymbolizer)[0];
```

Comment:
Can we be sure that the symbolizer name is always in the first item?

Reply:
yes